File: Data Structures & Algorithms/binary-search/submission-14.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length;

        while (l < r) {
            int m = l + ((r - l) / 2);

            if (nums[m] >= target) {
                r = m;
            } else {
                l = m + 1;
            }
        }

        return (l < nums.length && nums[l] == target) ? l : -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-14.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {} # val, index

        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in lookup:
                return [lookup[complement], i]
            else:
                lookup[nums[i]] = i